gctscript/README.md

```diff
@@ -0,0 +1,241 @@
+
+Join our slack to discuss all things related to GoCryptoTrader! [GoCryptoTrader Slack](https://join.slack.com/t/gocryptotrader/shared_invite/enQtNTQ5NDAxMjA2Mjc5LTc5ZDE1ZTNiOGM3ZGMyMmY1NTAxYWZhODE0MWM5N2JlZDk1NDU0YTViYzk4NTk3OTRiMDQzNGQ1YTc4YmRlMTk)
+
+## Current Features for gctscript package
+
++ Execute scripts
++ Terminate scripts
++ Autoload scripts on bot startup
++ Current Exchange features supported:
+  + Enabled Exchanges
+  + Enabled currency pairs
+  + Account information
+  + Query Order
+  + Submit Order
+  + Cancel Order
+  + Ticker
+  + Orderbook
+
+## How to use
+
+##### Prerequisites
+
+To Enable database logging support you must have an active migrated database by following the [database setup guide](../database/README.md)
+
+##### Configuration
+
+The gctscript configuration struct is currently: 
+```shell script
+type Config struct {
+	Enabled       bool          `json:"enabled"`
+	ScriptTimeout time.Duration `json:"timeout"`
+	AllowImports  bool          `json:"allow_imports"`
+	AutoLoad      []string      `json:"auto_load"`
+	Verbose       bool          `json:"Verbose"`
+}
+```
+
+With an example configuration being:
+
+```sh
+ "gctscript": {
+  "enabled": true,
+  "timeout": 600000000,
+  "allow_imports": true,
+  "auto_load": [],
+  "debug": false
+ },
+```
+##### Script Control
++ You can autoload scripts on bot start up by placing their name in the "auto_load" config entry
+  ```shell script
+  "auto_load": ["one","two"]
+  ```
+  This will look in your GoCryptoTrader data directory in a folder called "scripts" for files one.gct and two.gct and autoload them
++ Manual control of scripts can be done via the gctcli command with support for the following:
+  
+  - Start/Execute:
+  ```shell script
+    gctcli gctscript execute <scriptname> <pathoverride>
+    gctcli gctscript execute "timer.gct" "~/gctscript"
+  
+    {
+      "status": "ok",
+      "data": "timer.gct executed"
+    }
+  ```
+  - Stop:
+  ```shell script
+    gctcli gctscript stop <uuid>
+    gctcli gctscript stop 821bd73e-02b1-4974-9463-874cb49f130d
+  
+    {
+      "status": "ok",
+      "data": "821bd73e-02b1-4974-9463-874cb49f130d terminated"
+    }
+  ```
+  - Status:
+  ```shell script
+    gctcli gctscript status 
+  
+    {
+      "status": "ok",
+      "scripts": [
+        {
+          "uuid": "821bd73e-02b1-4974-9463-874cb49f130d",
+          "name": "timer.gct",
+          "next_run": "2019-11-14 13:11:40.224919456 +1100 AEDT m=+91.062103259"
+        }
+      ]
+    }
+  ```
+  - Read file:
+  ```shell script
+    gctcli gctscript read <filename>
+    gctcli gctscript read "timer.gct"
+  
+    {
+      "status": "ok",
+      "script": {
+      "name": "timer.gct",
+      "path": "/home/andrew/.gocryptotrader/scripts"
+      },
+      "data": "fmt := import(\"fmt\")\nt := import(\"times\")\n\nname := \"run\"\ntimer := \"5s\"\n\nload := func() {\n\tfmt.printf(\"5s %s\\n\",t.now())\n}\n\nload()\n"
+    }
+   ```
+    - Query running script:
```

Comment:
Please also add an example to show autoload functionality now that we support it